src/internal/restgateway/restgateway.go

```diff
@@ -0,0 +1,112 @@
+package restgateway
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"google.golang.org/grpc"
+
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+
+	"github.com/pachyderm/pachyderm/v2/src/admin"
+	"github.com/pachyderm/pachyderm/v2/src/auth"
+	"github.com/pachyderm/pachyderm/v2/src/debug"
+	"github.com/pachyderm/pachyderm/v2/src/enterprise"
+	"github.com/pachyderm/pachyderm/v2/src/identity"
+	"github.com/pachyderm/pachyderm/v2/src/license"
+	"github.com/pachyderm/pachyderm/v2/src/proxy"
+	"github.com/pachyderm/pachyderm/v2/src/transaction"
+	"github.com/pachyderm/pachyderm/v2/src/version/versionpb"
+	"github.com/pachyderm/pachyderm/v2/src/worker"
+)
+
+func NewMux(ctx context.Context, grpcConn *grpc.ClientConn) http.Handler {
+
+	ctx = pctx.Child(ctx, "restgateway")
+
+	mux := runtime.NewServeMux(runtime.WithIncomingHeaderMatcher(func(s string) (string, bool) {
+		if s != "Content-Length" {
+			return s, true
+		}
+		return s, false
+	}))
+
+	err := pps.RegisterAPIHandler(ctx, mux, grpcConn)
```

Comment:
I recommend `if err := ...; err != nil` here so that you can't accidentally reuse the errors between all of these different sections.  Honestly, you could also:

```go
handlers := []func(context.Context, *runtime.ServeMux, *grpc.ClientConn)error{
   pps.RegisterAPIHandler,
   pfs.RegisterAPIHandler,
   ...
}

var errs errors
for _, f := range handlers {
   if err := f(ctx, mux, grpcConn); err != nil {
     errors.JoinInto(&errs, err)
   }
}
if errs != nil {
   return errs
}
```